src/App.tsx

```diff
@@ -1,11 +1,44 @@
+import { useState } from "react";
+
+import { DisplayCard } from "./components/DisplayCard/DisplayCard";
+import { fetchSingleCharacter } from "./utils/fetchFunctions";
+
 const App = () => {
-	return (
-		<>
-			<div>
-				<h1>Pokémon Battle</h1>
-			</div>
-		</>
-	);
+  const [nameInputValue, setNameInputValue] = useState<string>("");
+  const [entities, setEntities] = useState<any[]>([]);
+
+  const handleSubmit = async () => {
+    setEntities([...entities, await fetchSingleCharacter(nameInputValue)]);
+    setNameInputValue("");
+  };
+
+  return (
+    <>
+      <div>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
```

Comment:
You should be able to handle this `preventDefault` in the `handleSubmit` function, once you remove it from the button click as well.